src/uuid/builder/uuidbuilder.rs

```diff
@@ -183,52 +114,29 @@ mod tests {
     use super::*;
 
     #[test]
-    fn test_string_to_obj_conversions() {
-        let str1 = "01868381-1590-8000-cfe2-68135f43b363";
-        let uuid = Uuid::parse_str(str1).unwrap();
-        let str2 = uuid.to_string();
-        assert_eq!(str1, str2);
+    fn test_build_uuid8_with_instant() {
+            let instant = 0x18C684468F8u64; // Thu, 14 Dec 2023 12:19:23 GMT
+            let uprotocol_uuid = UUIDv8Builder::new().build_with_instant(instant);
+        
+        // instant, version (8) and counter (000) should show up in UUID
+        assert!(uprotocol_uuid.to_hyphenated_string().starts_with("018c6844-68f8-8000-"));
+        // variant is 0b10
+        assert_eq!(uprotocol_uuid.lsb >> 62, 0b10);
     }
 
     #[test]
     fn test_obj_to_string_conversions() {
-        let uuid_factory = UUIDv8Builder::new();
-        let uuid1 = uuid_factory.build();
-        let str1 = uuid1.to_string();
-        let uuid2 = Uuid::parse_str(&str1).unwrap();
-        assert_eq!(str1, uuid2.to_string());
+        let uuid1 = UUIDv8Builder::new().build();
+        let str1 = uuid1.to_hyphenated_string();
+        let uuid2 = uuid::Uuid::parse_str(&str1).unwrap();
+        assert_eq!(str1, uuid2.as_hyphenated().to_string());
     }
 
     #[test]
     fn test_uuid_for_constant_random() {
         let factory = UUIDv8Builder::new();
         let uuid1 = factory.build();
         let uuid2 = factory.build();
-        // assert_eq!(uuid1.to_fields_le().3, uuid2.to_fields_le().3); // Check that the "node" field (least significant 64 bits) is the same
         assert_eq!(uuid1.lsb, uuid2.lsb);
     }
```

Comment:
Another one:
```rust
    #[test]
    fn test_uuid_past_max_count_in_instant() {
        let instant = 0x18C684468F8u64; // Thu, 14 Dec 2023 12:19:23 GMT
        let builder = UUIDv8Builder::new();

        // build with the same instant MAX_COUNT amount of times
        for _ in 0..=MAX_COUNT {
            let _ = builder.build_with_instant(instant);
        }

        let uprotocol_uuid_past_max_count = builder.build_with_instant(instant);

        // same instant, version (8) and _saturated_ counter (FFF) should show up in UUID
        assert!(uprotocol_uuid_past_max_count.to_hyphenated_string().starts_with("018c6844-68f8-8fff-"));
        // variant is 0b10
        assert_eq!(uprotocol_uuid_past_max_count.lsb >> 62, 0b10);
    }
```

Reply:
FMPOV the handling of the next UUID being built for the same instant would be even more worth being tested :-)

Reply:
Hey, did you mean using the same builder, same instant twice?

That's what I did [over here](https://github.com/eclipse-uprotocol/uprotocol-rust/pull/9#discussion_r1428950767) in this comment.

This test was meant to check the counter saturation logic.